### DISCUSSION TITLE
Sudden change in default font size in plots

### DISCUSSION THREAD

**Discussion by GitHubUserWithoutAName** [mtex-toolbox/mtex#1098] [Ask Anything]
Dear MTEX community,
I just ran into a very strange problem. I use scripts and version control when evaluating data with MTEX to be able to reproduce anything I calculated precisely. This means that any output should be more or less deterministic.

Despite this, all of a sudden, the default font size in all my plots has changed (most notably IPF and PF plots, but also the scale bars in EBSD plots etc.). As a result, the proportions of any plots that my scripts produce do not match previous versions anymore. This is pretty problematic due to further processing I apply to the figures I produce: Nothing quite aligns anymore.

I can really not fathom how this happened: I am 100% certain that I did NOT change the default font size in mtex_settings, it is and has been (MTEX 5.6.1 on Matlab R2019a, on Windows 10):
`% default fontsize
% the next lines try to set the fontsize depending on the screen resolution
ppi = get(0,'ScreenPixelsPerInch');
fontSize = round(15 * ppi/100);`
Even if I try to manually adjust the font size (it is now 14 and used to be something around 12, I presume), the font size looks similar to my old plots, but the proportions just do not quite match up. This I cannot understand since mtex_settings rounds to full integer values... I also did not change my screen resolution.
I even tried rolling back my code to some older versions - it does not appear to be related to my code at all (in which I did not - actively - change anything, anyways).

The change occurred over the weekend (everything was fine on Friday), and I am at a complete loss right now: Is there any (hidden) setting besides the default font size in mtex_settings that may influence the default font size as well as how elements are scale / aligned in plots? Maybe a general MATLAB setting (I did not - actively - change any)? Or is it possible that automatic windows update messed with my system / opengl driver / something like that? A reboot of my PC as well as Matlab did not change anything for the better.

Does anybody have any idea how this could have happened? Thanks for any hints.

**Comment by kilir** [comment]
Hi,

does anything change to the window placement, i.e. do plots "snap" to window borders or similar? While this should not change font size, it may change the arrangement in the plot. Declaring size and position of the plots directly might help. Another thought would a font mysteriously gone missing over the weekend, something changed in mtex_settings, especially `setMTEXpref('outerPlotSpacing', 50); `setMTEXpref('innerPlotSpacing',50);` which still should not really affect font scaling.

Cheers,
Rüdiger

**Reply by GitHubUserWithoutAName** [reply]
Hi Rüdiger,

thanks for your swift reply! Some of the settings you suggested (outerPlotSpacing), I already set explicitly. I could not notice any snapping of the figures to borders or such. I will consider setting more parameters explicitly.

I did manage to "fix" the problem, or rather it fixed itself: It seems that this has something to do with whether I directly work at my office PC or via remote (where I have a different screen resolution) - after remotely restarting Matlab, everything went back to normal. Why and how this happens (the default font size - which I now added as an output variable - remained the same, regardless of how I access the PC) I still do not quite understand, since I had switched back and forth from direct / remote access often in the past months.

Thanks once again!

Edit: I can actually reproduce this: The Matlab instance I remotely started yesterday evening does totally fine. I just started another instance right here in front of my PC and the issue reappeared. I do know a workaround now, but it is still quite strange that this has never happened before...

Second edit: I guess you are right, Rüdiger, that this is a problem regarding figure size and position. It looks like I might need to explicitly declare figure positions for all my figures manually. I currently only declare figure sizes via the flags 'large', 'huge' etc.
